File: C/substr.c
#include <stdio.h>
#include <string.h>

int main() {
    char mainstr[100];
    printf("Digite uma string: ");
    fgets(mainstr, sizeof(mainstr), stdin);
    mainstr[strcspn(mainstr, "\n")] = '\0';

    char substr[50];
    printf("Digite a substring: ");
    fgets(substr, sizeof(substr), stdin);
    substr[strcspn(substr, "\n")] = '\0';

    int i, contador = 0;

    for (i = 0; i <= strlen(mainstr) - strlen(substr); i++) {
        if (mainstr[i] == substr[0]) {
            int j;
            for (j = 0; j < strlen(substr); j++) {
                if (mainstr[i + j] != substr[j]) {
                    break;
                    }
                }
            if (j == strlen(substr)) {
                contador++;
            }
        }
    }
    printf("A substring %s aparece %d vezes na string %s.", substr, contador, mainstr);
    return 0;
}

File: Go/substr.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Insira a string principal: ")
	mainstr, err := reader.ReadString('\n')
	if err != nil {
		fmt.Println("Erro ao ler a string principal:", err)
		return
	}
	mainstr = strings.TrimSpace(mainstr)
	fmt.Print("Insira a substring: ")
	substr, err := reader.ReadString('\n')
	if err != nil {
		fmt.Println("Erro ao ler a substring:", err)
		return
	}
	substr = strings.TrimSpace(substr)

	if len(substr) == 0 || len(mainstr) == 0 {
		fmt.Println("As strings não podem ser vazias.")
		return
	}
	if len(substr) > len(mainstr) {
		fmt.Println("A substring não pode ser maior que a string principal.")
		return
	}
	cont := 0
	for i := 0; i <= len(mainstr)-len(substr); i++ {
		if mainstr[i] == substr[0] {
			j := 0
			for j = 0; j < len(substr); j++ {
				if mainstr[i+j] != substr[j] {
					break
				}
			}
			if j == len(substr) {
				cont++
			}
		}
	}
	fmt.Printf("A substring '%s' aparece %d vez(es) na string principal '%s'.\n", substr, cont, mainstr)
}
